File: Sort/quick-sort.py
def sort(array, b, e):
    l = b
    r = e
    p = array[int((l + r) / 2)]
    while l <= r:
        while array[l] < p:
            l += 1
        while array[r] > p:
            r -= 1
        if l <= r:
            if l < r:
                array[l], array[r] = array[r], array[l]
            l += 1
            r -= 1
    if b < r:
        sort(array, b, r)
    if e > l:
        sort(array, l, e)

a = []
s = input()

for i in s.split(' '):
    a.append(int(i))


sort(a, 0, len(a) - 1)

print(a)

File: Sort/Quick_Sort.cs
using System;

namespace Base
{
    class Program
    {
        public static void Sort(ref int[] array, int b, int e)
        {
            int l = b;
            int r = e;
            int p = array[(l + r) / 2];
            while(l <= r)
            {
                while(array[l] < p)
                {
                    l ++;
                }
                while(array[r] > p)
                {
                    r --;
                }
                if(l <= r)
                {
                    if(l < r)
                    {
                        array[l] = array[l] + array[r];
                        array[r] = array[l] - array[r];
                        array[l] = array[l] - array[r];
                    }
                    l ++;
                    r --;
                }
            }
            if(b < r)
            {
                Sort(ref array, b, r);
            }
            if(e > l)
            {
                Sort(ref array, l, e);
            }
        }

        public static void Main()
        {
            string[] w = Console.ReadLine().Split(" ");
            int[] a = new int[w.Length];
            for(int i = 0; i < w.Length; i ++)
            {
                a[i] = Convert.ToInt32(w[i]);
            }
            Sort(ref a, 0, a.Length - 1);
            foreach(int i in a)
            {
                Console.Write(i);
                Console.Write(" ");
            }
            Console.ReadLine();
        }
    }
}